### DISCUSSION TITLE
Handling sanitization of arrays submitted as params

### DISCUSSION THREAD

**Discussion by mwender** [TCattd/HTMX-API-WP#8] [Ideas]
When submitting a `<select multple name="select_multiple_example[]"></select>` with multiple values selected, running that value through `sanitize_text_value()` in `HXWP_Render::sanitize_params()` returns an empty value.

To get things working, I was able to write my own filter to handle arrays:

```
/**
 * Handles sanitizing arrays of params.
 *
 * @param      string $sanitized   The parameter after it has been run through sanitize_text_field().
 * @param      mixed  $unsanitized  The unsanitized value.
 *
 * @return     mixed  The sanitized paramter.
 */
function sanitize_array_params( $sanitized, $unsanitized ){
  if( is_array( $unsanitized ) ){
    foreach( $unsanitized as $key => $value ){
      $values[$key] = sanitize_text_field( $value );
    }
    $sanitized = $values;
  }

  return $sanitized;
}
add_filter( 'hxwp/sanitize_param_value', 'sanitize_array_params', 10, 2 );
```
However, I'm wondering if `HXWP_Render::sanitize_params()` should handle this natively since `<select multiple>` is pretty standard HTML?

**Comment by EstebanForge** [comment]
Hi @mwender 

Can you try v0.1.15, please?
Let me know.

Esteban

**Reply by mwender** [reply]
v0.1.15 properly handles the scenario I outline above. Thanks @TCattd!

**Reply by EstebanForge** [reply]
Thanks for letting me know :)
